File: packages/usage-spec-kotlin/bin/main/usage_spec/Spec.kt
package usage_spec

data class SpecChoices(
    val values: List<String>,
)

data class SpecArg(
    val name: String,
    val help: String,
    val required: Boolean,
    val `var`: Boolean,
    val hide: Boolean,
    val default: List<String>,
    val choices: SpecChoices?,
)

data class SpecFlag(
    val short: String,
    val long: String,
    val help: String,
    val helpLong: String,
    val required: Boolean,
    val hide: Boolean,
    val global: Boolean,
    val count: Boolean,
    val `var`: Boolean,
    val negate: String,
    val deprecated: String,
    val default: List<String>,
    val defaultBool: Boolean?,
    val env: String,
    val arg: SpecArg?,
)

data class SpecCommand(
    val name: String,
    val help: String,
    val helpLong: String,
    val hide: Boolean,
    val deprecated: String,
    val aliases: List<String>,
    val subcommandRequired: Boolean,
    val flags: List<SpecFlag>,
    val args: List<SpecArg>,
    val cmds: List<SpecCommand>,
)

data class Spec(
    val name: String,
    val bin: String,
    val version: String,
    val about: String,
    val long: String,
    val usage: String,
    val flags: List<SpecFlag>,
    val args: List<SpecArg>,
    val cmds: List<SpecCommand>,
)

File: packages/usage-spec-go/spec.go
package usagespec

type SpecChoices struct {
	Values []string
}

type SpecArg struct {
	Name     string
	Help     string
	Required bool
	Var      bool
	Hide     bool
	Default  []string
	Choices  *SpecChoices
}

type SpecFlag struct {
	Short       string
	Long        string
	Help        string
	HelpLong    string
	Required    bool
	Hide        bool
	Global      bool
	Count       bool
	Var         bool
	Negate      string
	Deprecated  string
	Default     []string
	DefaultBool *bool
	Env         string
	Arg         *SpecArg
}

type SpecCommand struct {
	Name               string
	Help               string
	HelpLong           string
	Hide               bool
	Deprecated         string
	Aliases            []string
	SubcommandRequired bool
	Flags              []SpecFlag
	Args               []SpecArg
	Cmds               []SpecCommand
}

type Spec struct {
	Name    string
	Bin     string
	Version string
	About   string
	Long    string
	Usage   string
	Flags   []SpecFlag
	Args    []SpecArg
	Cmds    []SpecCommand
}

func BoolPtr(v bool) *bool {
	return &v
}
